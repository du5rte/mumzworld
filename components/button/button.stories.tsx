import type { Meta, StoryObj } from '@storybook/react';
import { fn } from '@storybook/test';

import { Button } from './button';
import Box from '../box';

// More on how to set up stories at: https://storybook.js.org/docs/writing-stories#default-export
const meta: Meta<typeof Button> = {
  title: 'Components/Button',
  component: Button,
  parameters: {
    // Optional parameter to center the component in the Canvas. More info: https://storybook.js.org/docs/configure/story-layout
    layout: 'centered',
  },
  // This component will have an automatically generated Autodocs entry: https://storybook.js.org/docs/writing-docs/autodocs
  tags: ['autodocs'],
  // More on argTypes: https://storybook.js.org/docs/api/argtypes
  argTypes: {
    // backgroundColor: { control: 'color' },
  },
  // Use `fn` to spy on the onClick arg, which will appear in the actions panel once invoked: https://storybook.js.org/docs/essentials/actions#action-args
  args: {
    title: 'Button',
    onPress: fn(),
  },
};

export default meta;

type Story = StoryObj<typeof Button>;

// More on writing stories with args: https://storybook.js.org/docs/writing-stories/args

export const Variations = () => (
  <Box flexDirection="row" gap="s">
    <Button title="Primary" variant="primary" />
    <Button title="Secondary" variant="secondary" />
    <Button title="Highlight" variant="highlight" />
  </Box>
);

export const Sizes = () => (
  <Box flexDirection="row" gap="s" alignItems="center">
    <Button title="Small" size="s" />
    <Button title="Medium" size="m" />
    <Button title="Large" size="l" />
    <Button title="X Large" size="xl" />
  </Box>
);

export const Primary: Story = {
  args: {
    variant: 'primary',
  },
};
export const Highlight: Story = {
  args: {
    variant: 'highlight',
  },
};

export const Secondary: Story = {
  args: {
    variant: 'secondary',
  },
};

export const Disabled: Story = {
  args: {
    disabled: true,
  },
};
