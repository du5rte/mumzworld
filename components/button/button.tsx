import React from 'react';
import {
  Pressable,
  GestureResponderEvent,
  NativeSyntheticEvent,
  NativeMouseEvent,
  PressableProps,
  Platform,
} from 'react-native';
import Animated, {
  useAnimatedStyle,
  useDerivedValue,
  useSharedValue,
} from 'react-native-reanimated';
import { I18nManager } from 'react-native';

import useTheme from '@/hooks/useTheme';
import { withMediumTiming, withFastTiming, withFastestTiming } from '@/styles/timings';
import { interactiveElevationChange } from '@/styles/shadow';

import Icon, { FeatherIconNames } from '../icon';

const AnimatedPressable = Animated.createAnimatedComponent(Pressable);

export const buttonVariants = ['primary', 'secondary', 'highlight', 'text'] as const;
export const buttonSizes = ['xs', 's', 'm', 'l', 'xl'] as const;
export const buttonIconPositions = ['left', 'right'] as const;
export const buttonShapes = ['square', 'rectangle', 'round', 'circle'] as const;

export type ButtonVariants = (typeof buttonVariants)[number];
export type ButtonSizes = (typeof buttonSizes)[number];
export type ButtonIconPositions = (typeof buttonIconPositions)[number];
export type ButtonShapes = (typeof buttonShapes)[number];

const paddings: Record<ButtonSizes, number> = {
  xs: 14,
  s: 16,
  m: 18,
  l: 24,
  xl: 32,
};

const sizes: Record<ButtonSizes, number> = {
  xs: 32,
  s: 40,
  m: 44,
  l: 56,
  xl: 64,
};

const borderRadius: Record<ButtonSizes, number> = {
  xs: 8,
  s: 10,
  m: 12,
  l: 15,
  xl: 24,
};

const fontSizes: Record<ButtonSizes, number> = {
  xs: 13,
  s: 14,
  m: 15,
  l: 16,
  xl: 17,
};

const iconSizes: Record<ButtonSizes, number> = {
  xs: 16,
  s: 18,
  m: 20,
  l: 22,
  xl: 24,
};

export interface ButtonProps extends PressableProps {
  title?: string;
  variant?: ButtonVariants;
  size?: ButtonSizes;
  shape?: ButtonShapes;
  disabled?: boolean;
  float?: boolean;
  color?: string;
  icon?: FeatherIconNames;
  iconPosition?: ButtonIconPositions;
  elevate?: number;
  onPress?: PressableProps['onPress'];
  onPressIn?: PressableProps['onPress'];
  onPressOut?: PressableProps['onPress'];
  onHoverIn?: PressableProps['onHoverIn'];
  onHoverOut?: PressableProps['onHoverOut'];
  entering?: any;
  exiting?: any;
}

export function Button(props: ButtonProps) {
  const {
    variant = 'primary',
    size = 'm',
    shape = 'rectangle',
    disabled,
    title,
    icon,
    iconPosition = I18nManager.isRTL ? 'right' : 'left',
    elevate,
    onPress,
    onPressIn,
    onPressOut,
    onHoverIn,
    onHoverOut,

    entering,
    exiting,
    // ...rest
  } = props;

  const theme = useTheme();

  const press = useSharedValue(false);
  const hover = useSharedValue(false);

  const animatedBackgroundColor = useDerivedValue(() => {
    if (disabled) return withFastTiming(theme.colors.inactiveBackground);
    if (variant === 'primary') {
      return withFastTiming(hover.value ? theme.colors.primaryHover : theme.colors.primary);
    }
    if (variant === 'highlight') {
      return withFastTiming(
        press.value
          ? theme.colors.highlightPress
          : hover.value
            ? theme.colors.highlightHover
            : theme.colors.highlight
      );
    }
    if (variant === 'secondary') {
      return withFastTiming(theme.colors.background);
    }
    return withMediumTiming(theme.colors.transparent);
  });

  const animatedTextColor = useDerivedValue(() => {
    if (disabled) return withFastestTiming(theme.colors.inactive);

    if (variant === 'highlight') {
      return withFastestTiming(
        press.value ? theme.colors.highlightPressContrast : theme.colors.contrast
      );
    }
    if (variant === 'secondary' || variant === 'text') {
      return withFastestTiming(
        press.value
          ? theme.colors.secondaryPress
          : hover.value
            ? theme.colors.secondary
            : theme.colors.contrast
      );
    }
    return withFastestTiming(press.value ? theme.colors.primaryPress : theme.colors.contrast);
  });

  const animatedBackgroundStyle = useAnimatedStyle(() => {
    return {
      backgroundColor: animatedBackgroundColor.value,
      borderRadius: borderRadius[size],
      ...(Platform.OS === 'ios' && {
        borderCurve: 'continuous',
      }),
      ...((shape === 'square' || shape === 'circle') && {
        width: sizes[size],
      }),
      ...((shape === 'round' || shape === 'circle') && {
        borderRadius: theme.borderRadii.round,
      }),
      // Needs to have a backgroundColor for the shadow to work on iOS
      ...((variant === 'secondary' || 'elevate' in props) &&
        interactiveElevationChange(elevate || 10, { press, hover })),
      ...(Platform.OS === 'web' && {
        cursor: 'pointer',

        ...(disabled && {
          cursor: 'auto',
        }),
      }),
    };
  });

  const animatedWrapperStyle = useAnimatedStyle(() => {
    return {
      // flex: 1,
      flexDirection: 'row',
      justifyContent: 'center',
      alignItems: 'center',
      flexShrink: 1,
      flexGrow: 0,
      height: sizes[size],
      ...((shape === 'rectangle' || shape === 'round') && {
        paddingHorizontal: paddings[size],
      }),
      ...((shape === 'square' || shape === 'circle') && {
        width: sizes[size],
      }),
      ...((shape === 'round' || shape === 'circle') && {
        borderRadius: theme.borderRadii.round,
      }),
      gap: 8,
    };
  });

  const animatedIconStyle = useAnimatedStyle(() => {
    return {
      color: animatedTextColor.value,
    };
  });

  const animatedTextStyle = useAnimatedStyle(() => {
    return {
      fontFamily: 'Inter-Medium',
      ...(Platform.OS === 'web' && {
        fontFamily: 'Inter, sans-serif',
        fontWeight: '500',
      }),
      fontSize: fontSizes[size],
      numberOfLines: 1,
      color: animatedTextColor.value,
      ...(Platform.OS === 'web' && {
        userSelect: 'none',
        WebkitUserSelect: 'none',
      }),
    };
  });

  const handlePress = (event: GestureResponderEvent) => {
    if (onPress) onPress(event);
  };
  const handlePressIn = (event: GestureResponderEvent) => {
    press.value = true;
    if (onPressIn) onPressIn(event);
  };
  const handlePressOut = (event: GestureResponderEvent) => {
    press.value = false;
    if (onPressOut) onPressOut(event);
  };
  const handleHoverIn = (event: NativeSyntheticEvent<NativeMouseEvent>) => {
    hover.value = true;
    if (onHoverIn) onHoverIn(event);
  };
  const handleHoverOut = (event: NativeSyntheticEvent<NativeMouseEvent>) => {
    hover.value = false;
    if (onHoverOut) onHoverOut(event);
  };

  return (
    <AnimatedPressable
      accessibilityRole="button"
      accessibilityState={!disabled ? { selected: true } : {}}
      accessibilityLabel={title}
      onPress={handlePress}
      onPressIn={handlePressIn}
      onPressOut={handlePressOut}
      onHoverIn={handleHoverIn}
      onHoverOut={handleHoverOut}
      disabled={disabled}
      testID={`button-${variant}${disabled ? '-disabled' : ''}`}
      style={[animatedWrapperStyle, animatedBackgroundStyle]}
      entering={entering}
      exiting={exiting}>
      {icon && iconPosition === 'left' && (
        <Animated.Text style={animatedIconStyle}>
          <Icon name={icon} size={iconSizes[size]} />
        </Animated.Text>
      )}

      {title && <Animated.Text style={animatedTextStyle}>{title}</Animated.Text>}

      {icon && iconPosition === 'right' && (
        <Animated.Text style={animatedIconStyle}>
          <Icon name={icon} size={iconSizes[size]} />
        </Animated.Text>
      )}
    </AnimatedPressable>
  );
}

export default Button;
