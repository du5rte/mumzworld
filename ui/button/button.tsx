import { PressableProps } from 'react-native';
import { View, styled, withStaticProperties } from '@tamagui/core';

import Icon, { FeatherIconNames } from '@/components/icon';

import Text from '../text';

/**
 * Read guide: [How to build a button](https://tamagui.dev/docs/guides/how-to-build-a-button)
 */

const ButtonFrame = styled(View, {
  name: 'Button',

  flexDirection: 'row',
  justifyContent: 'center',
  alignItems: 'center',
  gap: '$s',
  animation: 'fast',
  pressStyle: {
    scale: 0.95,
  },

  variants: {
    size: {
      xs: {
        height: '$xs',
        paddingHorizontal: 14,
        borderRadius: 8,
      },
      s: {
        height: '$s',
        paddingHorizontal: 16,
        borderRadius: 10,
      },
      m: {
        height: '$m',
        paddingHorizontal: 18,
        borderRadius: 12,
      },
      l: {
        height: '$l',
        paddingHorizontal: 24,
        borderRadius: 15,
      },
      xl: {
        height: '$xl',
        paddingHorizontal: 32,
        borderRadius: 24,
      },
    },
    variant: {
      primary: {
        backgroundColor: '$primary',
        hoverStyle: {
          backgroundColor: '$primaryHover',
        },
      },
      secondary: {
        backgroundColor: '$background',
      },
      highlight: {
        backgroundColor: '$highlight',
        hoverStyle: {
          backgroundColor: '$highlightHover',
        },
        pressStyle: {
          backgroundColor: '$highlightPress',
        },
      },
    },
  } as const,
  defaultVariants: {
    size: 'm',
    variant: 'primary',
  },
});

const ButtonText = styled(Text, {
  name: 'ButtonText',

  fontWeight: '500',
  numberOfLines: 1,
  userSelect: 'none',

  animation: 'fast',

  variants: {
    size: {
      xs: {
        fontSize: 13,
      },
      s: {
        fontSize: 14,
      },
      m: {
        fontSize: 15,
      },
      l: {
        fontSize: 16,
      },
      xl: {
        fontSize: 17,
      },
    },
    variant: {
      primary: {
        color: '$contrast',
      },
      secondary: {
        color: '$primary',
        // hoverStyle: {
        //   color: '$secondary',
        // },
        // pressStyle: {
        //   color: '$secondaryPress',
        // },
      },
      highlight: {
        color: '$contrast',
        // pressStyle: {
        //   color: '$highlightPressContrast',
        // },
      },
    },
  } as const,
});

// export const Button = withStaticProperties(ButtonFrame, {
//   Text: ButtonText,
// });

export interface ButtonProps extends PressableProps {
  text?: string;
  // variant?: ButtonVariants;
  // size?: ButtonSizes;
  // shape?: ButtonShapes;
  disabled?: boolean;
  icon?: FeatherIconNames;
  // iconPosition?: ButtonIconPositions;
  elevate?: number;
  onPress?: PressableProps['onPress'];
}

export function Button(props: ButtonProps) {
  const { text, size = 'm', variant = 'primary', disabled, textProps, onPress, ...rest } = props;

  return (
    <ButtonFrame {...{ size, variant, disabled }} {...rest}>
      <ButtonText {...{ size, variant, disabled }} {...textProps}>
        {text}
      </ButtonText>
    </ButtonFrame>
  );
}

export default Button;
