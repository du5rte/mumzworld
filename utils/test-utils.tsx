import React, { PropsWithChildren } from 'react';
import { render } from '@testing-library/react-native';
import { ThemeProvider } from '@shopify/restyle';
import { lightTheme } from '@/styles/themes';
import '@/locales/i18next-jest';

const AllTheProviders = ({ children }: PropsWithChildren) => {
  return <ThemeProvider theme={lightTheme}>{children}</ThemeProvider>;
};

const customRender: typeof render = (ui, options?) =>
  render(ui, { wrapper: AllTheProviders, ...options });

// re-export everything
export * from '@testing-library/react-native';

// override render method
export { customRender as render };
