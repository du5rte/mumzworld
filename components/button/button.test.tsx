import { fireEvent, render, screen } from '@/utils/test-utils';

import { Button } from './button';

describe('button', () => {
  const title = 'Test Button';

  it('should render text', () => {
    render(<Button title={title} />);
    expect(screen.getByText(title)).toBeDefined();
  });

  it('should call onPress when pressed', () => {
    const title = 'Test Button';

    const onPressMock = jest.fn();

    render(<Button title={title} onPress={onPressMock} />);

    fireEvent.press(screen.getByText(title));

    expect(onPressMock).toHaveBeenCalled();
  });
});

/**
 * Example of testing multiple design variations of a component using Storybook stories
 * This is a more intuitive way of testing that the component matches the design
 *
 * Alternatively this could also be done using visual regression testing
 * https://storybook.js.org/docs/writing-tests/visual-testing

import { composeStories } from '@storybook/react';

import * as stories from './button.stories';

const { Primary, Secondary, Highlight } = composeStories(stories);

describe('button stories', () => {
  it('should render Primary story correctly', () => {
    render(<Primary />);
    expect(screen.toJSON()).toMatchSnapshot();
  });

  it('should render Secondary story correctly', () => {
    render(<Secondary />);
    expect(screen.toJSON()).toMatchSnapshot();
  });

  it('should render Highlight story correctly', () => {
    render(<Highlight />);
    expect(screen.toJSON()).toMatchSnapshot();
  });
});

*/
