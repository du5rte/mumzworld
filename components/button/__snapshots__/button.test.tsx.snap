// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`button should render correctly 1`] = `
<View
  accessibilityLabel="Hello World!"
  accessibilityRole="button"
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": true,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  testID="button-primary"
>
  <View
    collapsable={false}
    jestAnimatedStyle={
      {
        "value": {
          "alignItems": "center",
          "flexDirection": "row",
          "flexGrow": 0,
          "flexShrink": 1,
          "gap": 8,
          "height": 44,
          "justifyContent": "center",
          "paddingHorizontal": 18,
        },
      }
    }
    style={
      {
        "alignItems": "center",
        "flexDirection": "row",
        "flexGrow": 0,
        "flexShrink": 1,
        "gap": 8,
        "height": 44,
        "justifyContent": "center",
        "paddingHorizontal": 18,
      }
    }
  >
    <View
      collapsable={false}
      jestAnimatedStyle={
        {
          "value": {
            "backgroundColor": "hsl(0, 0%, 0%)",
            "borderCurve": "continuous",
            "borderRadius": 12,
            "cursor": "pointer",
          },
        }
      }
      style={
        {
          "backgroundColor": "hsl(0, 0%, 0%)",
          "borderCurve": "continuous",
          "borderRadius": 12,
          "bottom": 0,
          "cursor": "pointer",
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    />
    <Text
      collapsable={false}
      jestAnimatedStyle={
        {
          "value": {
            "color": "hsl(0, 0%, 100%)",
            "fontFamily": "Inter-Medium",
            "fontSize": 15,
            "numberOfLines": 1,
          },
        }
      }
      style={
        {
          "color": "hsl(0, 0%, 100%)",
          "fontFamily": "Inter-Medium",
          "fontSize": 15,
          "numberOfLines": 1,
        }
      }
    >
      Hello World!
    </Text>
  </View>
</View>
`;

exports[`button stories should render Highlight story correctly 1`] = `
<View
  accessibilityLabel="Button"
  accessibilityRole="button"
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": true,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  testID="button-highlight"
>
  <View
    collapsable={false}
    jestAnimatedStyle={
      {
        "value": {
          "alignItems": "center",
          "flexDirection": "row",
          "flexGrow": 0,
          "flexShrink": 1,
          "gap": 8,
          "height": 44,
          "justifyContent": "center",
          "paddingHorizontal": 18,
        },
      }
    }
    style={
      {
        "alignItems": "center",
        "flexDirection": "row",
        "flexGrow": 0,
        "flexShrink": 1,
        "gap": 8,
        "height": 44,
        "justifyContent": "center",
        "paddingHorizontal": 18,
      }
    }
  >
    <View
      collapsable={false}
      jestAnimatedStyle={
        {
          "value": {
            "backgroundColor": "hsl(216, 100%, 48%)",
            "borderCurve": "continuous",
            "borderRadius": 12,
            "cursor": "pointer",
          },
        }
      }
      style={
        {
          "backgroundColor": "hsl(216, 100%, 48%)",
          "borderCurve": "continuous",
          "borderRadius": 12,
          "bottom": 0,
          "cursor": "pointer",
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    />
    <Text
      collapsable={false}
      jestAnimatedStyle={
        {
          "value": {
            "color": "hsl(0, 0%, 100%)",
            "fontFamily": "Inter-Medium",
            "fontSize": 15,
            "numberOfLines": 1,
          },
        }
      }
      style={
        {
          "color": "hsl(0, 0%, 100%)",
          "fontFamily": "Inter-Medium",
          "fontSize": 15,
          "numberOfLines": 1,
        }
      }
    >
      Button
    </Text>
  </View>
</View>
`;

exports[`button stories should render Primary story correctly 1`] = `
<View
  accessibilityLabel="Button"
  accessibilityRole="button"
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": true,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  testID="button-primary"
>
  <View
    collapsable={false}
    jestAnimatedStyle={
      {
        "value": {
          "alignItems": "center",
          "flexDirection": "row",
          "flexGrow": 0,
          "flexShrink": 1,
          "gap": 8,
          "height": 44,
          "justifyContent": "center",
          "paddingHorizontal": 18,
        },
      }
    }
    style={
      {
        "alignItems": "center",
        "flexDirection": "row",
        "flexGrow": 0,
        "flexShrink": 1,
        "gap": 8,
        "height": 44,
        "justifyContent": "center",
        "paddingHorizontal": 18,
      }
    }
  >
    <View
      collapsable={false}
      jestAnimatedStyle={
        {
          "value": {
            "backgroundColor": "hsl(0, 0%, 0%)",
            "borderCurve": "continuous",
            "borderRadius": 12,
            "cursor": "pointer",
          },
        }
      }
      style={
        {
          "backgroundColor": "hsl(0, 0%, 0%)",
          "borderCurve": "continuous",
          "borderRadius": 12,
          "bottom": 0,
          "cursor": "pointer",
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    />
    <Text
      collapsable={false}
      jestAnimatedStyle={
        {
          "value": {
            "color": "hsl(0, 0%, 100%)",
            "fontFamily": "Inter-Medium",
            "fontSize": 15,
            "numberOfLines": 1,
          },
        }
      }
      style={
        {
          "color": "hsl(0, 0%, 100%)",
          "fontFamily": "Inter-Medium",
          "fontSize": 15,
          "numberOfLines": 1,
        }
      }
    >
      Button
    </Text>
  </View>
</View>
`;

exports[`button stories should render Secondary story correctly 1`] = `
<View
  accessibilityLabel="Button"
  accessibilityRole="button"
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": true,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  testID="button-secondary"
>
  <View
    collapsable={false}
    jestAnimatedStyle={
      {
        "value": {
          "alignItems": "center",
          "flexDirection": "row",
          "flexGrow": 0,
          "flexShrink": 1,
          "gap": 8,
          "height": 44,
          "justifyContent": "center",
          "paddingHorizontal": 18,
        },
      }
    }
    style={
      {
        "alignItems": "center",
        "flexDirection": "row",
        "flexGrow": 0,
        "flexShrink": 1,
        "gap": 8,
        "height": 44,
        "justifyContent": "center",
        "paddingHorizontal": 18,
      }
    }
  >
    <View
      collapsable={false}
      jestAnimatedStyle={
        {
          "value": {
            "backgroundColor": "hsl(0, 0%, 100%)",
            "borderCurve": "continuous",
            "borderRadius": 12,
            "cursor": "pointer",
            "shadowColor": "hsl(0, 0%, 0%)",
            "shadowOffset": {
              "height": 4,
              "width": 0,
            },
            "shadowOpacity": 0.085,
            "shadowRadius": 7,
          },
        }
      }
      style={
        {
          "backgroundColor": "hsl(0, 0%, 100%)",
          "borderCurve": "continuous",
          "borderRadius": 12,
          "bottom": 0,
          "cursor": "pointer",
          "left": 0,
          "position": "absolute",
          "right": 0,
          "shadowColor": "hsl(0, 0%, 0%)",
          "shadowOffset": {
            "height": 4,
            "width": 0,
          },
          "shadowOpacity": 0.085,
          "shadowRadius": 7,
          "top": 0,
        }
      }
    />
    <Text
      collapsable={false}
      jestAnimatedStyle={
        {
          "value": {
            "color": "hsl(0, 0%, 0%)",
            "fontFamily": "Inter-Medium",
            "fontSize": 15,
            "numberOfLines": 1,
          },
        }
      }
      style={
        {
          "color": "hsl(0, 0%, 0%)",
          "fontFamily": "Inter-Medium",
          "fontSize": 15,
          "numberOfLines": 1,
        }
      }
    >
      Button
    </Text>
  </View>
</View>
`;
